mlflow/pyfunc/__init__.py

```diff
@@ -1581,16 +1594,19 @@ def batch_predict_fn(pdf, params=None):
 
         elif env_manager == _EnvManager.LOCAL:
             if is_spark_connect:
```

Comment:
If we update the condition to be `is_spark_connect and not should_spark_connect_use_nfs` then we don't need below if else statement. But I'm fine keeping it here as well :)